### DISCUSSION TITLE
Events @change on radio button of Formkit Not working

### DISCUSSION THREAD

**Discussion by yogendrarr** [formkit/formkit#1372] [General]
<FormKit
                v-if="question.inputType === 'radio'"
                v-model="question.response"
                type="radio"
                :label="question.questionName"
                :options="question.optionsArray"
                :name="question.questionCode"
                validation="required"
                :classes="{
                  outer: 'fk-inline',
                }"
                @click="handleParentQuestionChange(index, question.response)"
              />

- which events should use for calling functions on radio button

**Comment by fenilli** [comment]
FormKit is not a UI library, so you don't need it for almost any case at all to have `v-model` or `@change/@click`.

So it depends on what you're doing, as to why it doesn't have, its because we use `@input` for all input events, but v-model is already a mutating event so no need for it too if using it.

**Reply by yogendrarr** [reply]
i dont get it, i am using @input cause i am performing some operation on radio button. so using@input giving me little bit lag to get my response how to fix this

**Reply by fenilli** [reply]
It all depends on what exactly you're doing and why are you using `v-model` and `@` events to begin with, as to `@input` giving a lag response is because commits are async by a debounce so they are not exactly perfectly in sync, and neither would be `@change` in that case, if you want the direct input you can use `@inputRaw`.

But like I said before FormKit itself doesn't need for you to use either, you can just do:
```html
<FormKit type="form" :value="{ name: 'yogendrarr' }">
  <FormKit type="text" label="Name" name="name" />
</FormKit>
```

That would just work, FormKit knows name is a child of the form, and form knows a child with name exists so it passes its value down.

So can you elaborate on what your object is instead, so maybe w can find a better solution?
